src/Currencies/SimpleCurrencies.php

```diff
@@ -0,0 +1,91 @@
+<?php
+
+namespace Money\Currencies;
+
+use Money\Currencies;
+use Money\Currency;
+use Money\Exception\UnknownCurrencyException;
+
+/**
+ * Cache the result of currency checking.
+ *
+ * @author George Mponos <gmponos@gmail.com>
+ */
+final class SimpleCurrencies implements Currencies
```

Comment:
Most probably you may disaggree with this name but I could not come with any better. Any thoughts